File: Original C# Code/GBase_Console_MGilliland/GBase_Console_MGilliland/GBaseExceptions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace GBase_Console_MGilliland
{
    /// <summary>
    /// An exception thrown when a command is not followed by the correct specifier 
    /// </summary>
    class GBaseCommandMustBeFollowedByException : Exception
    {
        public int Location { get; set; }

        public GBaseCommandMustBeFollowedByException(String commandName, String followedBy, int location) 
            : base("The command: \"" + commandName.ToUpper() + "\" must be followed by " + followedBy)
        {
            Location = location;    // An estimate of the location of the error, since characters are by pixel, this doesn't really work
        }
    }

    /// <summary>
    /// An exception thrown when a command is unrecognized
    /// </summary>
    class GBaseDoesNotRecognizeCommandException : Exception
    {
        public GBaseDoesNotRecognizeCommandException(String commandName)
            : base("\""+commandName + "\" is not recognized as a GBase command")
        {
        }
    }

    /// <summary>
    /// An exception thrown when a ROW command does not have attribute value pairs in a "with"..."in" clause
    /// </summary>
    class GBaseRowCommandMustHaveWithClauseException : Exception
    {
        public GBaseRowCommandMustHaveWithClauseException()
            : base("GBase ROW command must have a WITH clause specifying attr:value pairs")
        {
        }
    }

    /// <summary>
    /// An exception thrown when a ROW command does not have attribute value pairs in a "that" clause
    /// </summary>
    class GBaseModCommandMustHaveThatClauseException : Exception
    {
       public GBaseModCommandMustHaveThatClauseException()
            : base("GBase MOD command must have a THAT clause specifying attr:value pairs")
        {
        }
    }

    /// <summary>
    /// An excpetion thrown when attribute value pairs are incorrectly given
    /// </summary>
    class GBaseAttrValuePairsMustBeOfFormException : Exception
    {
        public GBaseAttrValuePairsMustBeOfFormException()
            : base("GBase attr value pairs must be specified as [attr]:[value]")
        {
        }
    }

    /// <summary>
    /// A general GBase specific exception with just a message
    /// </summary>
    class GBaseGeneralException : Exception
    {
        public GBaseGeneralException(String message)
            : base(message)
        {
        }
    }
}

File: Source/GBaseExceptions.py
#!/usr/bin/env python

# Created by Michael Gilliland
# Date: Mon May  7 19:44:17 EDT 2012
# 
# 

""" Exception for when a command is not followed by something"""
class GBaseCommandMustBeFollowedByException(Exception):
	def __init__(self, commandName, followedBy):
		self.message = 'The command: \"' + commandName.upper() + '\" must be followed by ' + followedBy
	def __str__(self):
		return repr(self.message)

""" Exception for when an unrecognized command is given """
class GBaseDoesNotRecognizeCommandException(Exception):
	def __init__(self, commandName):
		self.message = '\"' + commandName + '\" is not recognized as a GBase command'
        def __str__(self):
                return repr(self.message)

""" Exception for when a ROW command does not have a WITH Clause """
class GBaseRowCommandMustHaveWithClauseException(Exception):
	def __init__(self):
		self.message = "GBase ROW command must have a WITH clause specifying attr:value pairs"
        def __str__(self):
                return repr(self.message)

""" Exception for when a MOD command does not have a THAT Clause """
class GBaseModCommandMustHaveThatClauseException(Exception):
	def __init__(self):
		self.message = "GBase MOD command must have a THAT clause specifying attr:value pairs"
	def __str__(self):
		return repr(self.message)

""" Exception for when attr:value pairs are wrong """
class GBaseAttrValuePairsMustBeOfFormException(Exception):
	def __init__(self):
		self.message = "GBase attr value pairs must be specified as [attr]:[value]"
	def __str__(self):
		return repr(self.message)

""" Exception for general problems """
class GBaseGeneralException(Exception):
	def __init__(self, message):
		self.message = message
	def __str__(self):
		return repr(self.message)
